lib/LedgerSMB/PSGI.pm

```diff
@@ -199,6 +208,17 @@ sub _run_old {
     }
 }
 
+sub _fxrate_app {
+    my $env = shift;
+    my $session = $env->{'psgix.session'};
+    if ( $session && $session->{valid}) {
```

Comment:
This does work, psgix.session is written behind the scene by a memory write, so that's why it seems to appear from nowhere. Tested and proven correct.